### ISSUE TITLE
✅ 🧩 🐞 WooCommerce Notice widget style doesn't work - [ED-13218]

### ISSUE THREAD

**Issue by codeorlov** [elementor/elementor#24862]
### Prerequisites

- [X] I have searched for similar issues in open and closed tickets and cannot find a duplicate.
- [X] I have troubleshooted my issue, and it still exists against the latest stable version of Elementor.

### Description

WooCommerce notices style doesn't work globally. 
![image](https://github.com/elementor/elementor/assets/130235685/9c3fb3ef-c921-456f-880b-07ae182285cc)

### Steps to reproduce

1. Reproduced all steps from official article https://elementor.com/help/woocommerce-notice-widget/

### Expected behavior

1. WooCommerce notices can be stylizeble via Elementor accordion official documentation.

### Isolating the problem

- [X] This bug happens when only the Elementor (and Elementor Pro) plugins are active.
- [X] This bug happens with the Hello Elementor theme active.
- [X] I can reproduce this bug consistently by following the steps I described above.

### Elementor System Info

<details>
<summary>Click to reveal</summary>

````txt

== Server Environment ==
	Operating System: Linux
	Software: Apache
	MySQL version: mariadb.org binary distribution v10.11.6-MariaDB-1:10.11.6+maria~ubu2004
	PHP Version: 8.2.14
	PHP Memory Limit: 512M
	PHP Max Input Vars: 5000
	PHP Max Post Size: 64M
	GD Installed: Yes
	ZIP Installed: Yes
	Write Permissions: All right
	Elementor Library: Connected

== WordPress Environment ==
	Version: 6.4.2
	Site URL: https://dev.integrio.com.ua
	Home URL: https://dev.integrio.com.ua
	WP Multisite: No
	Max Upload Size: 64 MB
	Memory limit: 256M
	Max Memory limit: 512M
	Permalink Structure: /%category%/%postname%/
	Language: en-US
	Timezone: Europe/Kyiv
	Debug Mode: Inactive

== Theme ==
	Name: Hello Elementor
	Version: 3.0.0
	Author: Elementor Team
	Child Theme: No

== User ==
	Role: administrator
	WP Profile lang: en_US
	User Agent: Mozilla/5.0 (Windows NT 10.0; Win64; x64) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/120.0.0.0 Safari/537.36

== Active Plugins ==
	Classic Editor
		Version: 1.6.3
		Author: WordPress Contributors

	Code Snippets
		Version: 3.6.2
		Author: Code Snippets Pro

	Code Snippets Pro (Premium)
		Version: 3.6.3
		Author: Code Snippets Pro

	Elementor
		Version: 3.18.3
		Author: Elementor.com

	Elementor Pro
		Version: 3.18.3
		Author: Elementor.com

	Loco Translate
		Version: 2.6.6
		Author: Tim Whitlock

	Scroll Styler
		Version: 1.1
		Author: jablonczay

	ShopBuilder Pro – Elementor WooCommerce Builder Addons
		Version: 1.1.1
		Author: RadiusTheme

	ShopBuilder – Elementor WooCommerce Builder Addons
		Version: 2.1.1
		Author: RadiusTheme

	WooCommerce
		Version: 8.5.1
		Author: Automattic


== Features ==
	Custom Fonts: 0
	Custom Icons: 0

== Integrations ==
	
	woocommerce: Active


== Elementor Experiments ==
	Optimized DOM Output: Active
	Improved Asset Loading: Active
	Improved CSS Loading: Active
	Inline Font Icons: Inactive
	Additional Custom Breakpoints: Active
	admin_menu_rearrangement: Inactive by default
	Flexbox Container: Active
	Upgrade Swiper Library: Active
	Grid Container: Inactive
	Default to New Theme Builder: Inactive
	Hello Theme Header & Footer: Inactive
	Editor Top Bar: Active
	Optimized Gutenberg Loading: Inactive
	Build with AI: Inactive
	Landing Pages: Inactive
	Nested Elements: Active
	Lazy Load Background Images: Inactive
	Optimize Image Loading: Active
	Pages Panel: Inactive by default
	Global Style Guide: Active
	Page Transitions: Active
	Notes: Inactive
	Form Submissions: Active
	Scroll Snap: Inactive
	Menu: Active
	Taxonomy Filter: Inactive by default


== Log ==
	


== Elementor - Compatibility Tag ==
	
	Elementor Pro: Compatible
	ShopBuilder Pro – Elementor WooCommerce Builder Addons: Compatible
	ShopBuilder – Elementor WooCommerce Builder Addons: Compatible

== Elementor Pro - Compatibility Tag ==
	
	ShopBuilder Pro – Elementor WooCommerce Builder Addons: Compatible
	ShopBuilder – Elementor WooCommerce Builder Addons: Compatible

````

</details>

### Agreement

- [X] I agree that my issue may be closed without action if it doesn't meet all the requirements.

**Comment by nicholaszein** [comment]
Hello @codeorlov! 

Thank you for submitting your issue! 🙏 

We are aware of this issue and our team is already working on a fix. We can’t say exactly when it'll be sorted out, but we kindly ask for your patience. Thanks for understanding! 😉

Best regards

**Comment by mister-mst** [comment]
I have the same problem, I was looking to see if I was the only one.

**Comment by mateitudor** [comment]
Just wait 6 years, they'll get around to it!

**Comment by lensbreak** [comment]
> Just wait 6 years, they'll get around to it!

exactly my worry as with all the other bugs that are still open, most like since 6+ months

**Comment by 79ho3ein** [comment]
> @mister-mst @mateitudor @nicholaszein @codeorlov Can you please make sure that the CSS has been regenerated (Elementor > Tools: Regenerate Files button).

@hein-obox, WooCommerce has changed its notifications in version 8.5.0 which caused this problem. Even in other plugins and tools like bricks, this problem has arisen.

**Comment by giannisgx89** [comment]
WooCommerce is now using a Font Library for the notices icons and several WooCommerce notices show a weird box icon now instead of the actual "info" or "error" icons. Only the "checkmark" icon shows up correctly.

**Comment by hein-obox** [comment]
@giannisgx89 Could you share a screenshot to make sure that we understand the problem correctly?

I know that WooCommerce has updated the notice icons. That change has been made outside of Elementor.

@nicholaszein Possibly we could consider a control inside our Notice controls to change the specific Notice icon to give the user more control over the Notice behavior?

**Comment by giannisgx89** [comment]
@hein-obox 

Hello,

Sure, here is the WooCommerce Status Report in case you need it: https://pastebin.com/raw/C4FiL1zc

I also forgot to mention that I am using a custom code snippet (generated with ChatGPT) to display a “Free Shipping” notice on the cart page and the elementor’s mini cart sidebar: https://pastebin.com/raw/2jEVbG7E . The issue also happens even when i have the snippet disabled. Like i mentioned above only the success notice shows the correct icon.
The console doesn't show any errors when I am inspecting the page.

![526zSlo](https://github.com/elementor/elementor/assets/5347711/b89697fa-2f86-461a-b080-3c262b35bf45)

**Comment by hein-obox** [comment]
@giannisgx89 From what I can see you aren't using any Elementor functionality for this.
This Notice seems to be directly generated from PHP, and excludes the Elementor widgets.

About the expected icons and unexpected icons, can you please send multiple screenshots, to indicate in what cases it works as expected and in what cases it doesn't works as expected. I am not sure what you are expecting.

These are the icons that I am seeing on my localhost: 

![Screenshot 2024-03-04 at 13 31 28](https://github.com/elementor/elementor/assets/94352322/1e0b2a4f-afc0-4cb9-94ce-ed5f418dc2d5)

**Comment by giannisgx89** [comment]
You are absolutely correct and I'm sorry for the confusion. After I cleared my cdn cache regenerated CSS and added content: "\e016"; in .woocommerce-info::before class, the icon appeared correctly.

**Comment by hein-obox** [comment]
@giannisgx89 No problem. Good luck with your project.

**Comment by fazanash** [comment]
Is it solved?

**Comment by digimarkup** [comment]
This issue is with the Elementor Hello theme. I'm using the Hello child theme and I am not using the Notification widget. My notifications look like  @giannisgx89 screenshot. When I switch to a none Elementor theme, I see the new WooCommerce notification. I've been having this problem for months. It has been almost a year. The new notifications use the wc-block-components-notice-banner class. The notifications on my site use the woocommerce-info class.

**Comment by giannisgx89** [comment]
I ended up using this css to fix the issue

.woocommerce-info::before {
	content: "\e016";
}

**Comment by digimarkup** [comment]
Ok, I'll give it a try. Thanks

**Comment by hein-obox** [comment]
@digimarkup This seems to be a WooCommerce product decision. Block-based themes will display the block notification, while other themes will show the other template. I don't think this is related to any Elementor decisions.

But I think it might be helpful if Elementor would add an option to customize the notice icon. Would that help you?

**Comment by digimarkup** [comment]
@hein-obox I don't think it's a block-based theme issue. With the WooCoommerce release right before 8.7, I could see the new notification style on my website that uses the Hello theme. After the 8.7 release, the old notifications came back. So the non-block-based themes are capable of showing the new notification design. 

If you look at the first image on this page you can see the new notification design. Elementor probably switched back to the old design to make it easier for people to design it with their widget. There needs to be an option that allows you to choose which notification version you want to use.

**Comment by nicholaszein** [comment]
Hello everyone!

It's been a while, but we are finally closing this issue after verifying it was working correctly with the latest version of Elementor (v3.20.4), Elementor Pro (3.20.3), and WooCommerce (v8.7.0). So...

### 📢 We're happy to announce that the issue you raised was resolved in Elementor Pro v3.19.0! 🥳 

✅ Feel free to check it out and update your plugin to the new version!

As for the ability to change the Notice icons, this is currently not possible due to compatibility with WooCommerce itself. However, if you think this should be a feature worth to implement, please join the discussion below, and cast your ⬆️ **Upvote**!
- #25709 

Thank you all for your feedback and contribution!

> _Check the changelog for more information:_
> [Changelog of Elementor and Elementor Pro](https://elementor.com/help/changelog-of-elementor/)

Cheers 🥂

**Comment by hein-obox** [comment]
@digimarkup You are right about the WooCommerce Notice styling in different WooCommerce versions. The block-style notices were introduced by WooCommerce in 8.5 and reversed in 8.6. This was changed by WooCommerce, and this isn't an Elementor issue.

See: https://developer.woocommerce.com/2024/02/15/woocommerce-8-6-product-blocks-order-attribution-updates/.

<img width="964" alt="Screenshot 2024-04-12 at 09 26 48" src="https://github.com/elementor/elementor/assets/94352322/fcdf97d9-c1e0-4e6c-967e-937d699ac45c">
